src/pkg/backup/details/details.go

```diff
@@ -628,15 +628,20 @@ func UpdateItem(item *ItemInfo, newLocPath *path.Builder) {
 	// contained in them.
 	var updatePath func(newLocPath *path.Builder)
 
-	switch item.infoType() {
-	case ExchangeContact, ExchangeEvent, ExchangeMail:
+	// Can't switch based on infoType because that's been unstable.
+	if item.Exchange != nil {
 		updatePath = item.Exchange.UpdateParentPath
-	case SharePointLibrary:
+	} else if item.SharePoint != nil {
+		// SharePoint used to store library items with the OneDriveItem ItemType.
+		// Start switching them over as we see them since there's no point in
+		// keeping the old format.
+		if item.SharePoint.ItemType == OneDriveItem {
+			item.SharePoint.ItemType = SharePointLibrary
+		}
+
 		updatePath = item.SharePoint.UpdateParentPath
-	case OneDriveItem:
+	} else if item.OneDrive != nil {
 		updatePath = item.OneDrive.UpdateParentPath
-	default:
```

Comment:
Do we need to keep this as an `else { return }`?

Reply:
yea, we should. good catch, thanks